### PR TITLE
[SYCLomatic] Enable CMake migration of CUDA_64_BIT_DEVICE_CODE

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Adds support for migration of CUDA_64_BIT_DEVICE_CODE to equivalent SYCL.

There are many if-condtions that check for CUDA_64_BIT_DEVICE_CODE so added rule and equivalent SYCL_64_BIT_DEVICE_CODE Cmake variable.

Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

File: clang/test/dpct/cmake_migration/case_035/input.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
+set(CUDA_64_BIT_DEVICE_CODE ON)
+
+set(CUDA_64_BIT_DEVICE_CODE OFF)
+
+if(CUDA_64_BIT_DEVICE_CODE)
```
Comment (the-slow-one) [inline clang/test/dpct/cmake_migration/case_035/input.cmake:5]: This case?
`if (NOT CUDA_64_BIT_DEVICE_CODE)`

Comment (daiyaan-ahmed6) [reply clang/test/dpct/cmake_migration/case_035/input.cmake:5]: Done.

**Review by tomflinda — APPROVED** [review]
LGTM

**Review by the-slow-one — APPROVED** [review]

**Comment by the-slow-one** [comment]
Looks good to me.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**